flow/otel_metrics/observables.go

```diff
@@ -99,6 +102,52 @@ func NewFloat64SyncGauge(meter metric.Meter, gaugeName string, opts ...metric.Fl
 	return &Float64SyncGauge{syncGauge: syncGauge}, nil
 }
 
+func buildFlowMetadataAttributes(flowMetadata *protos.FlowContextMetadata) metric.MeasurementOption {
+	return metric.WithAttributeSet(attribute.NewSet(
+		attribute.String(FlowNameKey, flowMetadata.FlowName),
+		attribute.String(SourcePeerType, flowMetadata.Source.Type.String()),
+		attribute.String(DestinationPeerType, flowMetadata.Destination.Type.String()),
+		attribute.String(SourcePeerName, flowMetadata.Source.Name),
+		attribute.String(DestinationPeerName, flowMetadata.Destination.Name),
+	))
+}
+
+type ContextAwareInt64SyncGauge struct {
+	Int64SyncGauge
+}
+
+func (a *ContextAwareInt64SyncGauge) Record(ctx context.Context, value int64, options ...metric.RecordOption) {
+	flowMetadata := shared.GetFlowMetadata(ctx)
+	if flowMetadata != nil {
+		options = append(options, buildFlowMetadataAttributes(flowMetadata))
+	}
+	a.Int64SyncGauge.Record(ctx, value, options...)
+}
+
+type ContextAwareFloat64SyncGauge struct {
+	Float64SyncGauge
```

Comment:
```suggestion
	metric.Float64Gauge
```
can we keep this wrapping interface instead of returning interface & then downcasting?

if not methods like Float64Gauge should return Float64SyncGauge. Probably should either way

Reply:
my bad, forgot to refactor